packages/core/src/factory.ts

```diff
@@ -5,38 +5,23 @@ import * as TaskEither from 'fp-ts/lib/TaskEither';
 import { configMergerFactory, PartialPrismConfig, PrismConfig } from '.';
 import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, PickRequired, ProblemJsonError } from './types';
 
-export function factory<Resource, Input, Output, Config, LoadOpts>(
+export function factory<Resource, Input, Output, Config>(
   defaultConfig: PrismConfig<Config, Input>,
-  defaultComponents: Partial<IPrismComponents<Resource, Input, Output, Config, LoadOpts>>,
+  defaultComponents: Partial<IPrismComponents<Resource, Input, Output, Config>>,
 ): (
   customConfig?: PartialPrismConfig<Config, Input>,
-  customComponents?: PickRequired<Partial<IPrismComponents<Resource, Input, Output, Config, LoadOpts>>, 'logger'>,
-) => IPrism<Resource, Input, Output, Config, LoadOpts> {
+  customComponents?: PickRequired<Partial<IPrismComponents<Resource, Input, Output, Config>>, 'logger'>,
+) => IPrism<Resource, Input, Output, Config> {
   const prism = (
     customConfig?: PartialPrismConfig<Config, Input>,
-    customComponents?: PickRequired<Partial<IPrismComponents<Resource, Input, Output, Config, LoadOpts>>, 'logger'>,
+    customComponents?: PickRequired<Partial<IPrismComponents<Resource, Input, Output, Config>>, 'logger'>,
   ) => {
     const components: PickRequired<
-      Partial<IPrismComponents<Resource, Input, Output, Config, LoadOpts>>,
+      Partial<IPrismComponents<Resource, Input, Output, Config>>,
       'logger'
     > = Object.assign({}, defaultComponents, customComponents);
-
-    // our loaded resources (HttpOperation objects, etc)
-    let resources: Resource[] = [];
-
     return {
-      get resources(): Resource[] {
-        return resources;
-      },
-
-      load: async (opts?: LoadOpts): Promise<void> => {
-        const { loader } = components;
-        if (opts && loader) {
-          resources = await loader.load(opts, defaultComponents.loader);
-        }
-      },
-
-      process: async (input: Input, c?: Config) => {
+      process: async (input: Input, resources: Resource[], c?: Config) => {
```

Comment:
Yes — Prism is now completely stateless and you need to supply the available resources every time you want to mock something.

The state will be held (if really necessary) by some external software (CLI, Studio, something else)